kernel/sched.c

```diff
@@ -1338,44 +1338,49 @@ struct k_thread *z_priq_rb_best(struct _priq_rb *pq)
 }
 
 #ifdef CONFIG_SCHED_MULTIQ
-# if (K_LOWEST_THREAD_PRIO - K_HIGHEST_THREAD_PRIO) > 31
-# error Too many priorities for multiqueue scheduler (max 32)
-# endif
 
 static ALWAYS_INLINE void z_priq_mq_add(struct _priq_mq *pq,
 					struct k_thread *thread)
 {
-	int priority_bit = thread->base.prio - K_HIGHEST_THREAD_PRIO;
+	int offset_priority = thread->base.prio - K_HIGHEST_THREAD_PRIO;
+	int priority_idx = BIT_ARRAY_IDX(offset_priority);
+	int priority_bit = BIT_ARRAY_OFFSET(offset_priority);
```

Comment:
Style: you repeat this construction in a few places.  Might be cleaner to define a utility that returns both a word pointer and a bit index for a given priority.

Reply:
Yep - a static (possibly inline) function in this file would do it.